### DISCUSSION TITLE
Quickest way to verify if two meshes intersect?

### DISCUSSION THREAD

**Discussion by fdinhSG** [pyvista/pyvista#5320] [Q&A] [answered]
Hey,

I need to verify whether or not two meshes intersect.

What I'm doing right now is to compute the intersection with `intersection = mesh1.boolean_intersection(mesh2)` and to verify that `intersection.n_points > 0`.

I realize that there's another function that I can use, `mesh.intersection(mesh2)`, but I don't know which is faster. Moreover, I don't really need to calculate the whole volume (or surface) intersection between the two, I just need to know if they do intersect.

Is there a quicker way to do this? Or can I interrupt the function execution in the moment I verify there's even a point in common?

Thanks!

**Comment by darikg** [comment, marked as answer]
there's [collision](https://docs.pyvista.org/version/stable/api/core/_autosummary/pyvista.PolyDataFilters.collision.html#pyvista-polydatafilters-collision) with contact_mode=first contact. (But that only checks face intersection, and won't catch the case where one surface is entirely contained inside the other.)

**Reply by fdinhSG** [reply]
Thanks! That was perfect. I'll keep in mind its limitation.
